### PR TITLE
Expose more stats for TaskDescriptorStorage

### PR DESCRIPTION
**Pull request by losipiuk**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java [hunk 1]
```diff
@@ -299,55 +343,125 @@ private void throwIfFailed()
...
+
+        public Stream<Long> getStagesReservedBytes()
+        {
+            return stagesReservedBytes.values().stream()
+                    .map(AtomicLong::get);
+        }
     }
 
-    private static class TaskDescriptorKey
+    public static class StorageStats
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java:366]: Curious, is it possible to expose a "managed record"?

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java (hunk 1 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java:407]: lets skip to until never

File: core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java [hunk 2]
```diff
@@ -197,36 +197,34 @@ public synchronized long getReservedBytes()
...
+
         private long reservedBytes;
         private RuntimeException failure;
 
         public void put(StageId stageId, int partitionId, TaskDescriptor descriptor)
         {
             throwIfFailed();
-            TaskDescriptorKey key = new TaskDescriptorKey(stageId, partitionId);
-            checkState(descriptors.putIfAbsent(key, descriptor) == null, "task descriptor is already present for key %s ", key);
+            checkState(descriptors.put(stageId, partitionId, descriptor) == null, "task descriptor is already present for key %s/%s ", stageId, partitionId);
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java:285]: would be nice to have putIfAbsent semantics not to currpt state if incorrect use

maybe descriptors.rowMap  putifabsent

File: README.md [hunk 3]
```diff
@@ -86,8 +86,8 @@ development. Use the following options to create a run configuration:
 
 * Main Class: `io.trino.server.DevelopmentServer`
 * VM Options: `-ea -Dconfig=etc/config.properties -Dlog.levels-file=etc/log.properties -Djdk.attach.allowAttachSelf=true`
-* Working directory: `$MODULE_DIR$`
-* Use classpath of module: `trino-server-dev`
+``* Working directory: `$MODULE_DIR$`
+``* Use classpath of module: `trino-server-dev`
```
Comment (findepi) [inline README.md:90]: revert?

File: core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java [hunk 4]
```diff
@@ -188,25 +200,92 @@ private synchronized void updateMemoryReservation(long delta)
...
+    synchronized long getReservedBytes()
     {
         return reservedBytes;
     }
 
+    @Managed
+    @Nested
+    public StorageStats getStats()
+    {
+        // this must same instance on each call.
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java:213]: maybe explain why

File: core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java [hunk 5]
```diff
@@ -297,5 +378,105 @@ private void throwIfFailed()
...
+                    .map(AtomicLong::get);
+        }
+    }
+
+    private record StorageStatsValue(
+            @Managed long queriesCount,
+            @Managed long stagesCount,
+            @Managed long reservedBytes,
+            @Managed long queryReservedBytesAvg,
+            @Managed long queryReservedBytesP50,
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java:399]: those `@Managed` unused?

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java (hunk 4 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java:213]: changed a comment a bit

**Pull request merged**